### DISCUSSION TITLE
논리곱 연산자(&&)와 논리합 연산자(||)의 단축 평가 동작 방식을 설명해주세요.

### DISCUSSION THREAD

**Discussion by publdaze** [publdaze/js-study#34] [타입변환과 단축 평가]
- 논리곱 연산자(&&)와 논리합 연산자(||)의 단축 평가 동작 방식을 설명해주세요.

**Comment by pipisebastian** [comment]
> 논리곱 연산자(&&)와 논리합 연산자(||)의 단축 평가 동작 방식을 설명해주세요.


논리합(||) 연산자는 좌항의 피연산자가 true일 경우, 우항의 피연산자를 평가하지 않고, 좌항의 피연산자를 반환합니다.
논리곱(&&) 연산자는 좌, 우 모두 true인지를 검사하기 때문에, 좌항이 false인경우 우항을 검사하지 않습니다.

**Comment by publdaze** [comment]
> - 논리곱 연산자(&&)와 논리합 연산자(||)의 단축 평가 동작 방식을 설명해주세요.
- 논리곱 연산자(&&)는 좌항의 피연산자가 false일 때 더이상 평가하지 않고 false 반환, true 면 우항도 이어서 평가
- 논리합 연산자(||)는 좌항의 피연산자가 true 일 때 더이상 평가하지 않고 true 반환, false 면 우항도 이어서 평가
